Public/Src/Tools/Tool.MsBuildGraphBuilder/MsBuildAssemblyLoader.cs

```diff
@@ -31,48 +32,56 @@ public sealed class MsBuildAssemblyLoader : IMsBuildAssemblyLoader
         private const string SystemCollectionsImmutable = "System.Collections.Immutable.dll";
         private const string SystemThreadingDataflow = "System.Threading.Tasks.Dataflow.dll";
 
-        // MsBuild.exe is not a required to be loaded but is required to be found
-        private const string MsBuildExe = "MsBuild.exe";
+        // MsBuild is not a required to be loaded but is required to be found
+        // Under DotNetCore there is not an msbuild.exe but a msbuild.dll
+        private string m_msbuild;
 
-        private static readonly string[] s_assemblyNamesToLoad = new[]
-            {
-                MsBuildExe,
-                MicrosoftBuild,
-                MicrosoftBuildFramework,
-                MicrosoftBuildUtilities,
-                SystemCollectionsImmutable,
-                SystemThreadingDataflow
-            };
+        private readonly string[] m_assemblyNamesToLoad;
 
         /// <summary>
         /// The expected public token for each required assembly
         /// </summary>
-        private static readonly Dictionary<string, string> s_assemblyPublicTokens = new Dictionary<string, string>
-        {
-            [MsBuildExe] = MSBuildPublicKeyToken,
-            [MicrosoftBuild] = MSBuildPublicKeyToken,
-            [MicrosoftBuildFramework] = MSBuildPublicKeyToken,
-            [MicrosoftBuildUtilities] = MSBuildPublicKeyToken,
-            [SystemCollectionsImmutable] = DotNetPublicToken,
-            [SystemThreadingDataflow] = DotNetPublicToken,
-        };
+        private readonly Dictionary<string, string> m_assemblyPublicTokens;
 
         private static ResolveEventHandler s_msBuildHandler;
 
         // Assembly resolution is multi-threaded
-        private readonly ConcurrentDictionary<string, Assembly> m_loadedAssemblies = new ConcurrentDictionary<string, Assembly>(5, s_assemblyNamesToLoad.Length, StringComparer.OrdinalIgnoreCase);
-
-        private static readonly Lazy<MsBuildAssemblyLoader> s_singleton = new Lazy<MsBuildAssemblyLoader>(() => new MsBuildAssemblyLoader());
+        private readonly ConcurrentDictionary<string, Assembly> m_loadedAssemblies;
 
-        private MsBuildAssemblyLoader()
+        public MsBuildAssemblyLoader(bool msBuildRuntimeIsDotNetCore)
         {
-        }
+            m_msbuild = msBuildRuntimeIsDotNetCore ? "MsBuild.dll" : "MsBuild.exe";

```

Comment:
Actual casing is `MSBuild`, for whenever this needs to run on case sensitive file systems.

Reply:
Will change

---
In reply to: [301352162](https://github.com/microsoft/BuildXL/pull/561#discussion_r301352162) [](ancestors = 301352162)